pallets/asset/src/lib.rs

```diff
@@ -1354,8 +1368,12 @@ decl_module! {
             let sender = ensure_signed(origin)?;
             let did = Context::current_identity_or::<Identity<T>>(&sender)?;
             ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
-            <Tokens<T>>::mutate(&ticker, |token| token.treasury_did = treasury_did);
-            Self::deposit_event(RawEvent::TreasuryDidSet(did, ticker, treasury_did));
+            let mut old_treasury = None;
+            <Tokens<T>>::mutate(&ticker, |token| {
```

Comment:
This function should be removed in favor of using the authorization stuff.

Reply:
Ahh I forgot, I was going to refactor this to be clear_treasury.